eng/build.ps1

```diff
@@ -372,6 +373,13 @@ function TestUsingRunTests() {
     $args += " --sequential"
     $args += " --include '\.IntegrationTests'"
     $args += " --include 'Microsoft.CodeAnalysis.Workspaces.MSBuild.UnitTests'"
+
+    if ($lspEditor) {
+      $args += " --testfilter FullyQualifiedName~Roslyn.VisualStudio.IntegrationTests.LanguageServerProtocol|Editor=LanguageServerProtocol"
```

Comment:
I went with both a separate namespace and additional trait name to enable multiple testing scenarios:
1.  Running selected existing integration tests under the LSP editor (uses trait to select which tests).
2.  Writing LSP only integration tests (uses separate namespace).
3.  Ensure that the tests in 2) do not run in normal integration test runs.

Reply:
I guess the trait is sufficient right, if you just put the trait on containing methods? My strong worry about the magic namespace is that we might rename or move something and not realize that tests aren't running anymore. Having the single method with the trait makes it a bit easier to know everything either works or doesn't.

Reply:
(see other comment though that the only test currently in it's own namespace I think is better just combined with the existing test which checks if it's in LSP mode or not...)

Reply:
And yes, we once had an integration test CI that was passing because whatever glob was looking for assemblies to run wasn't updated, and so it was running nothing. That was a "fun" discovery.

Reply:
@jasonmalinowski If we ever want to support running a test in LSP only context then we must have a separate namespace or something similar.  Otherwise I have no way to filter out LSP-only tests from the normal run (since the normal run includes shared tests with the LSP trait).

I can guarantee that there will be LSP only tests, if not gotodef then completion or another feature.  I think even for the gotodef test you mention I must have an LSP version and a non-LSP version, even if I share most of the code as they have different input data (window caption).

Reply:
I guess as long as we're using environment variables to control this, there's no reason you can just haven't a ConditionalFact that only runs if the environment variable is set. You can also have the test just check the environment variable in a test if you need to change a string.

Reply:
Ah!  Today I learned about conditional facts.  Ok, I'll use that in a followup instead.